terraform/ceph/cluster.tf

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+locals {
+  cluster_keys = toset([for app, obj in module.ceph_mon_config.config: obj.cluster-name])
+  clusters = {for key in local.cluster_keys: key => {
+    mons   = {for mon, obj in module.ceph_mon_config.config: mon => obj if obj.cluster-name == key}
+    osds   = {for osd, obj in module.ceph_osd_config.config: osd => obj if obj.cluster-name == key}
+    csis   = {for csi, obj in module.ceph_csi_config.config: csi => obj if obj.cluster-name == key}
+  }}
+}
```

Comment:
Here we are filtering the manifest config to find all ceph applications that are part of the same "cluster" defined by the `cluster-name`